### PR TITLE
Miscelaneous Javadoc

### PR DESCRIPTION
**Pull request by IMB11**
- Raycasts (HitResult classes)
- WorldSavePath constants
- ColorHelper `0xAABBGGRR`

### CODE REVIEW

File: mappings/net/minecraft/util/WorldSavePath.mapping [hunk 1]
```diff
@@ -1,6 +1,30 @@
 CLASS net/minecraft/class_5218 net/minecraft/util/WorldSavePath
+	COMMENT A collection of constants that contain the path of key files and folders in a save directory.
+	FIELD field_24181 Lnet/minecraft/class_5218;
+		COMMENT Stores the statistics of players that have played on the world.
+		COMMENT {@since 1.7.2}
```
Comment (apple502j) [inline mappings/net/minecraft/util/WorldSavePath.mapping]: No need for this, we didn't exist back then, and everything in the javadoc is specific to the current Minecraft version.

File: mappings/net/minecraft/util/hit/BlockHitResult.mapping [hunk 2]
```diff
@@ -1,4 +1,5 @@
 CLASS net/minecraft/class_3965 net/minecraft/util/hit/BlockHitResult
+	COMMENT When a raycast hits a block, an instance of this class is returned as the {@see HitResult hit result.}
```
Comment (apple502j) [inline mappings/net/minecraft/util/hit/BlockHitResult.mapping]: I believe this should be `@linkplain`, and dot should be outside the tag.

File: mappings/net/minecraft/util/hit/EntityHitResult.mapping [hunk 3]
```diff
@@ -1,8 +1,10 @@
 CLASS net/minecraft/class_3966 net/minecraft/util/hit/EntityHitResult
+	COMMENT When a raycast hits an entity, an instance of this class is created as the {@see HitResult hit result.}
```
Comment (apple502j) [inline mappings/net/minecraft/util/hit/EntityHitResult.mapping]: same

File: mappings/net/minecraft/util/hit/HitResult.mapping [hunk 4]
```diff
@@ -1,9 +1,18 @@
 CLASS net/minecraft/class_239 net/minecraft/util/hit/HitResult
+	COMMENT A hit result is returned when a raycast is created.
+	COMMENT
+	COMMENT <p>It contains information on what the ray hit (or didn't hit) and the position of the collision point.
+	COMMENT
+	COMMENT <p>To get the current raycast of the client {@see net.minecraft.client.MinecraftClient#crosshairTarget you can read the current crosshair target.}
```
Comment (apple502j) [inline mappings/net/minecraft/util/hit/HitResult.mapping]: same

File: mappings/net/minecraft/util/hit/HitResult.mapping [hunk 5]
```diff
@@ -1,9 +1,18 @@
...
 		ARG 1 pos
 	METHOD method_17783 getType ()Lnet/minecraft/class_239$class_240;
 	METHOD method_17784 getPos ()Lnet/minecraft/class_243;
+		COMMENT {@return the position where the raycast hit/}
 	METHOD method_24801 squaredDistanceTo (Lnet/minecraft/class_1297;)D
+		COMMENT {@return the distance between the position of the HitResult to an entity squared}
 		ARG 1 entity
 	CLASS class_240 Type
+		FIELD field_1332 Lnet/minecraft/class_239$class_240;
+		FIELD field_1333 Lnet/minecraft/class_239$class_240;
```
Comment (apple502j) [inline mappings/net/minecraft/util/hit/HitResult.mapping]: Hmm, what are these?

File: mappings/net/minecraft/util/math/ColorHelper.mapping [hunk 6]
```diff
@@ -33,23 +33,44 @@ CLASS net/minecraft/class_5253 net/minecraft/util/math/ColorHelper
 			COMMENT <p>The returned value is between {@code 0} and {@code 255} (both inclusive).
 			ARG 0 argb
 	CLASS class_8045 Abgr
+		COMMENT Contains color-related helper methods that use ABGR colors represented
+		COMMENT as {@code 0xAABBGGRR}
```
Comment (apple502j) [inline mappings/net/minecraft/util/math/ColorHelper.mapping]: Missing period.

File: mappings/net/minecraft/util/math/ColorHelper.mapping [hunk 7]
```diff
@@ -33,23 +33,44 @@ CLASS net/minecraft/class_5253 net/minecraft/util/math/ColorHelper
...
+		COMMENT
+		COMMENT <p>ABGR is mostly used in {@see net.minecraft.client.texture.NativeImage native image blending.}
```
Comment (apple502j) [inline mappings/net/minecraft/util/math/ColorHelper.mapping]: see above

File: mappings/net/minecraft/util/math/ColorHelper.mapping [hunk 8]
```diff
@@ -33,23 +33,44 @@ CLASS net/minecraft/class_5253 net/minecraft/util/math/ColorHelper
...
 		METHOD method_48347 getRed (I)I
+			COMMENT {@return the red value of {@code abgr}}
+			COMMENT
+			COMMENT <p>The returned value is between {@code 0} and {@code 255} (both inclusive).
 			ARG 0 abgr
 		METHOD method_48348 getBgr (I)I
+			COMMENT {@return The color without the alpha attached}
+			COMMENT
+			COMMENT <p>The format is {@code 0xBBGGRR}
 			ARG 0 abgr
```
Comment (apple502j) [inline mappings/net/minecraft/util/math/ColorHelper.mapping:77]: missing period

**Review by apple502j — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/util/hit/HitResult.mapping (hunk 5 above)
Comment (IMB11) [reply mappings/net/minecraft/util/hit/HitResult.mapping]: Odd, I didn't add those

File: mappings/net/minecraft/util/WorldSavePath.mapping [hunk 9]
```diff
@@ -1,6 +1,28 @@
 CLASS net/minecraft/class_5218 net/minecraft/util/WorldSavePath
+	COMMENT A collection of constants that contain the path of key files and folders in a save directory.
+	FIELD field_24181 Lnet/minecraft/class_5218;
+		COMMENT Stores the statistics of players that have played on the world.
+	FIELD field_24182 Lnet/minecraft/class_5218;
+		COMMENT Stores the states of all players that have played in the world.
+	FIELD field_24183 Lnet/minecraft/class_5218;
+		COMMENT Stores individual states of players that have played on this world.
+		COMMENT Only exists on worlds created before 1.7.6
```
Comment (apple502j) [inline mappings/net/minecraft/util/WorldSavePath.mapping:9]: Missing period.

File: mappings/net/minecraft/util/math/ColorHelper.mapping [hunk 10]
```diff
@@ -33,23 +33,44 @@ CLASS net/minecraft/class_5253 net/minecraft/util/math/ColorHelper
...
 			ARG 0 argb
 	CLASS class_8045 Abgr
+		COMMENT Contains color-related helper methods that use ABGR colors represented
+		COMMENT as {@code 0xAABBGGRR}.
+		COMMENT
+		COMMENT <p>ABGR is mostly used in {@linkplain net.minecraft.client.texture.NativeImage native image blending}.
 		METHOD method_48342 getAlpha (I)I
+			COMMENT {@return the alpha value of {@code abgr}}
+			COMMENT
+			COMMENT <p> The returned value is between {@code 0} and {@code 255} (both inclusive).
```
Comment (apple502j) [inline mappings/net/minecraft/util/math/ColorHelper.mapping:47]: Should not have space here, same below

**Review by apple502j — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/util/hit/HitResult.mapping [hunk 11]
```diff
@@ -1,9 +1,16 @@
...
+	COMMENT
+	COMMENT <p>It contains information on what the ray hit (or didn't hit) and the position of the collision point.
+	COMMENT
+	COMMENT <p>To get the current raycast of the client {@linkplain net.minecraft.client.MinecraftClient#crosshairTarget you can read the current crosshair target}.
 	FIELD field_1329 pos Lnet/minecraft/class_243;
 	METHOD <init> (Lnet/minecraft/class_243;)V
 		ARG 1 pos
 	METHOD method_17783 getType ()Lnet/minecraft/class_239$class_240;
 	METHOD method_17784 getPos ()Lnet/minecraft/class_243;
+		COMMENT {@return the position where the raycast hit/}
```
Comment (haykam821) [inline mappings/net/minecraft/util/hit/HitResult.mapping:12]:
```suggestion
		COMMENT {@return the position where the raycast hit}
```

File: mappings/net/minecraft/util/hit/BlockHitResult.mapping [hunk 12]
```diff
@@ -15,13 +16,22 @@ CLASS net/minecraft/class_3965 net/minecraft/util/hit/BlockHitResult
...
+		COMMENT <p>Moves the hit result to a different side of the block.
 		ARG 1 side
 	METHOD method_17780 getSide ()Lnet/minecraft/class_2350;
+		COMMENT {@return the side of the block that was hit by the raycast}
 	METHOD method_17781 isInsideBlock ()Z
+		COMMENT {@return {@code true} if the raycast point is inside the block}
 	METHOD method_29328 withBlockPos (Lnet/minecraft/class_2338;)Lnet/minecraft/class_3965;
+		COMMENT {@return the modified hit result}
+		COMMENT
+		COMMENT <p>Moves the hit result to a different block.
```
Comment (haykam821) [inline mappings/net/minecraft/util/hit/BlockHitResult.mapping:36]: The documentation here seems to indicate that the instance is modified and returned. However, the method naming implies that a new instance is created instead. Is the documentation correct here?

File: mappings/net/minecraft/util/hit/BlockHitResult.mapping [hunk 13]
```diff
@@ -15,13 +16,22 @@ CLASS net/minecraft/class_3965 net/minecraft/util/hit/BlockHitResult
...
 	METHOD method_17777 getBlockPos ()Lnet/minecraft/class_2338;
+		COMMENT {@return the position of the block that was hit by the raycast}
 	METHOD method_17778 createMissed (Lnet/minecraft/class_243;Lnet/minecraft/class_2350;Lnet/minecraft/class_2338;)Lnet/minecraft/class_3965;
 		ARG 0 pos
 		ARG 1 side
 		ARG 2 blockPos
 	METHOD method_17779 withSide (Lnet/minecraft/class_2350;)Lnet/minecraft/class_3965;
+		COMMENT {@return the modified hit result}
+		COMMENT
+		COMMENT <p>Moves the hit result to a different side of the block.
```
Comment (haykam821) [inline mappings/net/minecraft/util/hit/BlockHitResult.mapping:27]: The documentation here seems to indicate that the instance is modified and returned. However, the method naming implies that a new instance is created instead. Is the documentation correct here?

**Review by haykam821 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**